book/art/fullnode.bob

```diff
@@ -1,27 +1,30 @@
-             .---------------------------.
-             |  Fullnode                 |
-             |                           |
- .--------.  |  .------------------.     |
- |        |---->|                  |     |
- | Client |  |  | JSON RPC Service |     |
- |        |<----|                  |     |
- `----+---`  |  `------------------`     |
-      |      |     ^                     |    .------------------.
-      |      |     |  .----------------. |    | Validators       |
-      |      |     |  | Gossip Service +----->|                  |
-      |      |     |  `--------+-------` |    |  .------------.  |
-      |      |     |         ^ |         |    |  |            |  |
-      |      |     |         | v         |    |  | Upstream   |  |
-      |      |  .--+---.   .-+---.       |    |  | Validators |  |
-      |      |  | Bank |<--| TVU |<--------------+            |  |
-      |      |  `------`   `-----`       |    |  `------------`  |
-      |      |     ^                     |    |                  |
-      |      |     |                     |    |  .------------.  |
-      |      |  .--+--.   .-----------.  |    |  |            |  |
-      `-------->| TPU +-->| Broadcast +--------->| Downstream |  |
-             |  `-----`   | Service   |  |    |  | Validators |  |
-             |            `-----------`  |    |  |            |  |
-             |                           |    |  `------------`  |
-             `---------------------------`    |                  |
-                                              `------------------`
-
+             .---------------------------------.
+             |  Fullnode                       |
+             |                                 |
+ .--------.  |  .------------------.           |
+ |        |---->|                  |           |
+ | Client |  |  | JSON RPC Service |           |
+ |        |<----|                  |           |
+ `----+---`  |  `------------------`           |
+      |      |     ^                           |
+      |      |     |    .----------------.     |  .------------------.
+      |      |     |    | Gossip Service |<------>| Validators       |
+      |      |     |    `----------------`     |  |                  |
+      |      |     |                           |  |                  |
+      |      |     |                           |  |  .------------.  |
+      |      |  .--+---.  .------.  .--------. |  |  |            |  |
+      |      |  | Bank |<-|Replay|  |Network |<------+ Upstream   |  |
+      |      |  | Forks|  |Stage |  |Observer| |  |  | Validators |  |
+      |      |  `------`  `--+---`  `------+-` |  |  |            |  |
+      |      |   ^           |  ^          |   |  |  `------------`  |
+      |      |   | .------   |  |          |   |  |                  |
+      |      |   | |Gossip|<-`.-+-------.  |   |  |                  |
```

Comment:
Is there a GossipVotes struct?  If no, delete this. At most, we might have an arrow labeled "vote" going from ReplayStage to TPU, but we currently don't label arrows, so would be better to delete it than introduce confusion with more important data flow.

Reply:
Vote is not going directly from ReplayStage to TPU. All fullnodes are voting via Gossip service. The leader looks up the votes from the gossip service. I could not find a better way to depict this. Any thoughts?

Reply:
It's not too bad, but I'd still suggest leaving it out until we have some mechanism for labeling arrows.

Reply:
👍